buildSrc/src/main/java/org/elasticsearch/gradle/ConcatFilesTask.java

```diff
@@ -65,18 +67,26 @@ public void setHeaderLine(String headerLine) {
     @Optional
     public String getHeaderLine() { return headerLine; }
 
+    List<String> fileReadAllLines(File f, String encoding) {
+        try{
+            return Files.readAllLines(f.toPath(), Charset.forName(encoding));
+        }
+        catch (IOException e) {
+            return new ArrayList<>();
+        }
+    }
+
     @TaskAction
     public void concatFiles() throws IOException {
         final String encoding = "UTF-8";
         if (getHeaderLine() != null) {
             Files.write(target.toPath(), (getHeaderLine() + '\n').getBytes(encoding));
         }
+        assert getHeaderLine() == "Header";
 
         // To remove duplicate lines
         LinkedHashSet<String> uniqueLines = new LinkedHashSet<>();
-        for (File f : getFiles()) {
-            uniqueLines.addAll(Files.readAllLines(f.toPath(), Charset.forName(encoding)));
-        }
+        getFiles().getFiles().forEach(f -> uniqueLines.addAll(fileReadAllLines(f, encoding)));
```

Comment:
This fails with `NullPointerException` when `setFiles` is not used in the tests (I guess because the first `getFiles()` returns `null`?).